### PR TITLE
added inital socket mode document

### PR DESCRIPTION
**Pull request by stevengill**
First pass at adding a doc for socket mode. Pretty bare bones. What else should it include? 

### Category (place an `x` in each of the `[ ]`)

* [x] Document pages under `/docs`

## Requirements (place an `x` in each `[ ]`)

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

**Comment by stevengill** [comment]
I'm thinking I should mention the Socket Mode adapter in here

File: docs/_basic/socket_mode.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from slack_bolt.adapter.socket_mode import SocketModeHandler
+
+# Install the Slack app and get xoxb- token in advance
+app = App(token=os.environ["SLACK_BOT_TOKEN"])
+
+if __name__ == "__main__":
+    # export SLACK_APP_TOKEN=xapp-***
+    # export SLACK_BOT_TOKEN=xoxb-***
+    SocketModeHandler(app, os.environ["SLACK_APP_TOKEN"]).start()
+```
```
Comment (seratch) [inline docs/_basic/socket_mode.md:26]: How about mentioning available adapters this way? (please polish the sentences as necessary)

---

While we recommend using [the built-in Socket Mode adapter](https://github.com/slackapi/bolt-python/tree/main/slack_bolt/adapter/socket_mode/builtin), there are a few other 3rd party library based implementations. Here is the list of available adapters. 

|PyPI Project|Bolt Adapter|
|-|-|
|[slack_sdk](https://pypi.org/project/slack-sdk/)|[slack_bolt.adapter.socket_mode](https://github.com/slackapi/bolt-python/tree/main/slack_bolt/adapter/socket_mode/builtin)|
|[websocket_client](https://pypi.org/project/websocket_client/)|[slack_bolt.adapter.socket_mode.websocket_client](https://github.com/slackapi/bolt-python/tree/main/slack_bolt/adapter/socket_mode/websocket_client)|
|[aiohttp](https://pypi.org/project/aiohttp/) (asyncio-based)|[slack_bolt.adapter.socket_mode.aiohttp](https://github.com/slackapi/bolt-python/tree/main/slack_bolt/adapter/socket_mode/aiohttp)|
|[websockets](https://pypi.org/project/websockets/) (asyncio-based)|[slack_bolt.adapter.socket_mode.websockets](https://github.com/slackapi/bolt-python/tree/main/slack_bolt/adapter/socket_mode/websockets)|

To use the asyncio-based ones such as aiohttp, your app needs to be compatible with asyncio's async/await programming model. `AsyncSocketModeHandler` is available for running `AsyncApp` and its async middleware and listeners. 

```python
from slack_bolt.app.async_app import AsyncApp
# The default is the aiohttp based implementation
from slack_bolt.adapter.socket_mode.async_handler import AsyncSocketModeHandler

app = AsyncApp(token=os.environ["SLACK_BOT_TOKEN"])

async def main():
    handler = AsyncSocketModeHandler(app, os.environ["SLACK_APP_TOKEN"])
    await handler.start_async()

if __name__ == "__main__":
    import asyncio
    asyncio.run(main())
```

To learn how to use `AsyncApp`, check [Using Async](https://slack.dev/bolt-python/concepts#async) document and [examples](https://github.com/slackapi/bolt-python/tree/main/examples).

**Comment by seratch** [comment]
@stevengill Just added document to the python-slack-sdk's PR. Let me know in the PR if you have comments.
* https://github.com/slackapi/python-slack-sdk/pulls
* https://github.com/slackapi/python-slack-sdk/blob/v3.2-socket-mode/docs-src/socket-mode/index.rst

File: docs/_basic/socket_mode.md (hunk 1 above)
Comment (stevengill) [reply docs/_basic/socket_mode.md:26]: Added to the other PR!

**Comment by stevengill** [comment]
Closing in favor of #200

**Pull request closed without merging**